### PR TITLE
Updated Hunter version for M1 support

### PR DESCRIPTION
**Pull request by GaroRobe**
Updated Hunter reference for both x64 and arm64 (M1) support on MasOS

### CODE REVIEW

**Review by kamilsa — APPROVED** [review]

File: cmake/Hunter/init.cmake [hunk 1]
```diff
@@ -31,7 +31,7 @@ set(
 include(${CMAKE_CURRENT_LIST_DIR}/HunterGate.cmake)
 
 HunterGate(
-    URL  "https://github.com/soramitsu/soramitsu-hunter/archive/tags/v0.23.257-soramitsu13.tar.gz"
-    SHA1 "8c88d8bac2dee771926b84a6e28642a267f88ea8"
+    URL "https://github.com/GaroRobe/soramitsu-hunter/archive/2ee42292f05a12c4c48bdb524a6f5051eb1db8e2.tar.gz"
```
Comment (xDimon) [inline cmake/Hunter/init.cmake:34]: Please, do not merge it into master before sotramitsu-hunter has been updated.

File: cmake/functions.cmake [hunk 2]
```diff
@@ -9,8 +9,7 @@ function(addtest test_name)
   add_executable(${test_name} ${ARGN})
   addtest_part(${test_name} ${ARGN})
   target_link_libraries(${test_name}
-      GTest::main
-      GMock::main
+  GTest::gmock_main
```
Comment (xDimon) [inline cmake/functions.cmake:12]: Looks unformatted. Add tab please, as everywhere.

File: cmake/Hunter/init.cmake (hunk 1 above)
Comment (GaroRobe) [reply cmake/Hunter/init.cmake:34]: Well, the purpose is to test Linux and MacOS builds. Not to merge yet.

File: cmake/functions.cmake (hunk 2 above)
Comment (GaroRobe) [reply cmake/functions.cmake:12]: Ok

**Review by xDimon — APPROVED** [review]

**Review by kamilsa — APPROVED** [review]

**Pull request merged**